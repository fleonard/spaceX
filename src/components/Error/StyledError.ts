import styled from "styled-components/macro";

import { Alert } from "@material-ui/lab";

const StyledAlert = styled(Alert)`
  width: 100%;
`;

export { StyledAlert };
