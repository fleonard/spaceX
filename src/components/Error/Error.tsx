import React from "react";
import { AlertTitle } from "@material-ui/lab";

import { CenteredContainer } from "../../globalStyles";

import { StyledAlert } from "./StyledError";

interface ErrorProps {
  text: string | null;
}

const Error: React.FC<ErrorProps> = ({ text }) => {
  return (
    <CenteredContainer>
      <StyledAlert severity="error">
        <AlertTitle>Error</AlertTitle>
        <strong>{text}</strong>
      </StyledAlert>
    </CenteredContainer>
  );
};

export default Error;
