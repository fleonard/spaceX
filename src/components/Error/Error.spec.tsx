import React from "react";
import { render } from "@testing-library/react";
import "jest-styled-components";

import Error from "./Error";

describe("Dragon Component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Error text={"There's been an issue with your data."} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the dragon list", async () => {
    const { getByText } = render(
      <Error text={"There's been an issue with your data."} />
    );

    expect(getByText("There's been an issue with your data.")).toBeDefined();
  });
});
