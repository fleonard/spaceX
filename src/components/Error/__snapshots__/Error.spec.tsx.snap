// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragon Component renders correctly 1`] = `
.c0 {
  display: grid;
  padding: 30px 0;
  place-items: center;
}

.c1 {
  width: 100%;
}

<div
  class="c0"
>
  <div
    class="MuiPaper-root MuiAlert-root MuiAlert-standardError c1 MuiPaper-elevation0"
    role="alert"
  >
    <div
      class="MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message"
    >
      <div
        class="MuiTypography-root MuiAlertTitle-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Error
      </div>
      <strong>
        There's been an issue with your data.
      </strong>
    </div>
  </div>
</div>
`;
